src/state-manager/TransactionQueue.ts

```diff
@@ -460,6 +535,16 @@ class TransactionQueue {
               /* prettier-ignore */ if (logFlags.error && logFlags.verbose) this.mainLogger.error(`broadcast_finalstate data == null`)
               continue
             }
+            // validate corresponding tell sender
+            if (_sender == null ||  _sender.id == null) {
+              /* prettier-ignore */ if (logFlags.error) this.mainLogger.error(`broadcast_finalstate invalid sender for data: ${data.accountId}, sender: ${JSON.stringify(_sender)}`)
+              continue
+            }
+            const isValidFinalDataSender = this.factValidateCorrespondingTellFinalDataSender(queueEntry, _sender.id)
```

Comment:
in broadcast_data we are using configContext.p2p.useFactCorrespondingTell condition, why we are not using here?

Reply:
this is a great point,  in this case I am certain that switched over now with no going back,  will get that config removed in a follow up.